### DISCUSSION TITLE
Denomination Test Online

### DISCUSSION THREAD

**Discussion by Marti-C** [jspsych/jsPsych#1183] [Q&A]
Hi everyone! We are trying to build a denomination test with Jspsych. We aim at testing 2 year olds and the test should be accessible to anyone online. 
Ideally our test should be composed by: 
-	46 stimuli images (to be orally named) ranked by AoA (max duration of exposure for each stimulus 30 seconds).
-	Each stimulus is divided by a fixation point (2 seconds).
-	Two different sounds (1 second): one presenting the stimulus and one presenting the fixation point. 
Each stimulus can be ended pressing a key on the keyboard (in our case “s” for correct denomination (Sì - in Italian), “n” for incorrect denomination and “space” if parents are dubious) in order to record reaction times. RTs will also be verified post-test by a RA through video recording. The video recording should be ideally part of the digital test (not a Meet or Skype call).
We are also using AlterVista as an internet site for our experiment. 
We have some issues with JsPsych: 
-	We can’t manage the stimulus plug-in for JsPsych at the same time of the audio plug-in. We tried with some plug-in (found in the forum) as: audio-keybord-response.js + image-keybord-response.js; stim_audio_association; multi-stim-multi-response.js. 
But they didn’t work. 
-	We also want to record a video or audio track of the experiment, but we can’t find a plug-in that allow us to do this. We tried with audio_record plug-in but the program read it as a task and not as a response. We noticed that a plug-in like this is not yet available, is it right?
-	Moreover, the speed of our test suffers a lot poor internet connection. 
Can anyone help us with these issues?
 Thanks very much in advance.
Cheers, 
Martina

**Comment by giuconte** [comment]
Hi Martina,

I think I can give you a little help here, 

**We can’t manage the stimulus plug-in for JsPsych at the same time of the audio plug-in**
To solve this issue, I suggest you have a look at the [jspsychophysics](https://jspsychophysics.hes.kyushu-u.ac.jp/) plug-in. Works fine for me.

**record a video or audio track of the experiment, but we can’t find a plug-in that allows us to do this**
You should have a look at this [support audio recording](https://github.com/jspsych/jsPsych/issues/494). I have tried the html-audio-response and it works fairly well for my needs.

**Moreover, the speed of our test suffers a lot poor internet connection.**
the stimuli are usually pre-loaded prior to the start of the experiment (when the webpage is loading) and the results are uploaded only at the very end (when the experimental task is over).
As far as I understand, during the experimental session, the internet connection is not used so it shouldn't produce any slowing. With a slow internet connection, I experienced difficulty in downloading all the stimuli and thus in starting the experiment but no slowing in between. Reducing the number and quality of images or audio files (if you have any) helps a lot.

Best,

Giulio.

**Comment by Marti-C** [comment]
Thank you very much Giulio for these advices! I'll try and I'll let you know.

**Comment by Marti-C** [comment]
Hi Giulio! 
Unfortunately I have to inform you that your suggestions didn't work for our necessities. 
I think because specific plugins that we need aren't yet available. 

Maybe you can help me with a plugin that allow us to standby the experiment. We tried with:
jsPsych.pauseExperiment();
jsPsych.resumeExperiment();
but it can't allow us to pause the experiment during the task. 

Thanks very much in advance.
Cheers,
Martina

**Reply by giuconte** [reply]
The two functions you listed might be handy to pause the experiment for a predetermined amount of time between one trial and the other
(see https://github.com/jspsych/jsPsych/blob/master/examples/pause-unpause.html)
if this is not what you want I'm afraid I haven't understood exactly what do you mean by "pausing" during the task
Giulio.

**Reply by Marti-C** [reply]
Hi Giulio, thank you so much for your help. I'm sorry, I think I didn't express myself so well. 
I'll try to explain you: our test is a timeline of stimuli that proceeds automatically without ever stopping to the end. However, it is designed for children who may need to stop sometimes.
Thus, we are looking for a plug-in that allows our patients "stop" and "resume" the experiment every time they want and every moment they need, pressing a key. So these "pauses" shouldn't be preset but chosen by the subjects. I don't know if a function like this exists, but it might be very usefull.
Thank you so much in any case.

**Reply by giuconte** [reply]
Hi,

I don't think you need a plug-in for this. I have done it the way I suggested to you before. If you press key "p" once will pause the experiment, if you press "p" again it will resume. It is working pretty well for me. You can add the snippet below to as "on finish" after your stimuli, you only have to add the key "p" to the allowed choices

                  on_finish: function(data){
                    if (jsPsych.pluginAPI.compareKeys(data.key_press,80)) {
                      jsPsych.pauseExperiment();
                      jsPsych.pluginAPI.getKeyboardResponse({
                        callback_function:function(){jsPsych.resumeExperiment()},
                        valid_responses: 'p',
                        rt_method: 'performance',
                        persist: true
                      });
                    }
